### ISSUE TITLE
training FBNet on CIFAR-10

### ISSUE THREAD

**Issue by sl-zeng** [JunrQ/NAS#15]
Hi JunrQ,

Thanks for your work, it helps a lot. When I trained the FBNet on CIFAR-10, the accuracy began to quickly drop at epoch 66. And when it comes to epoch 71, the accuracy is about 0.1 and both loss 
and ce are nan.  Is there something wrong or normal?

By the way, the lowest loss is about 8.5 at epoch 27 and then it gets higher.

Thanks!

**Comment by JunrQ** [comment]
I don't know which code you use for training. 

**If your latency is very big, you can test with a smaller alpha.**

The pytorch version should't have the problem, when i train cifar10 with pytorch, the acc reach 0.9 very fast.

But the MXNet version may have the problem you describe when training with big dataset ...and i haven't find the bug yet...

**Comment by sl-zeng** [comment]
I am running the pytorch version code.

The original latency is 438.04 with loss 9.99. Later at epoch 67, the latency is 593 and acc is 0.83 with loss 9.73. 

Is the latency data consistent with yours?

**Comment by JunrQ** [comment]
I don't remember the numbers.

Maybe i should do some experiments and share my log.

I think the problem you describe may be caused by a very small temperature, which makes gumbel_softmax overflow.

**Comment by ChengTsang** [comment]
> Hi JunrQ,
> Thanks for your work, it helps a lot. When I trained the FBNet on CIFAR-10, the accuracy began to quickly drop at epoch 66. And when it comes to epoch 71, the accuracy is about 0.1 and both loss
> and ce are nan.  Is there something wrong or normal?
> By the way, the lowest loss is about 8.5 at epoch 27 and then it gets higher.
> Thanks!

I've been in this situation too.

**Comment by JunrQ** [comment]
I run `train_cifar10.py` with following command

```shell
python train_cifar10.py --batch-size $[128*2] --gpus 0,1 --log-frequence 50
```
I add hyperparameters in log, and upload it.

        NAS/fbnet-pytorch/cifar10.alpha.0.01.init_lat.438.log

**Comment by sl-zeng** [comment]
Thanks a lot, I updated the code and everything seems working fine now. 
Maybe something wired with my project. 

Best Regards!

**Comment by andyhahaha** [comment]
I use exactly the same command and hyperparameter, but still get nan after few epochs.
Why is that happened?

**Comment by iamwangyabin** [comment]
I also got nan with only few epoch.
`IP:192.168.1.105 [1017 11:20:30 @model.py:254] Epoch[3] Batch[600] Speed: 116.755600 samples/sec loss: 7.42444 acc: 0.37574 ce: 1.66188 lat: 282.41042
IP:192.168.1.105 [1017 11:20:57 @model.py:254] Epoch[3] Batch[700] Speed: 118.149047 samples/sec loss: 7.40012 acc: 0.37723 ce: 1.65843 lat: 280.81454
IP:192.168.1.105 [1017 11:21:24 @model.py:254] Epoch[3] Batch[800] Speed: 120.087542 samples/sec loss: 7.37590 acc: 0.37880 ce: 1.65459 lat: 279.25515
IP:192.168.1.105 [1017 11:21:50 @model.py:254] Epoch[3] Batch[900] Speed: 120.278566 samples/sec loss: 7.35230 acc: 0.38035 ce: 1.65090 lat: 277.73209
IP:192.168.1.105 [1017 11:22:17 @model.py:254] Epoch[3] Batch[1000] Speed: 120.583996 samples/sec loss: 7.32869 acc: 0.38206 ce: 1.64675 lat: 276.24322
IP:192.168.1.105 [1017 11:22:43 @model.py:254] Epoch[3] Batch[1100] Speed: 120.124030 samples/sec loss: 7.30691 acc: 0.38325 ce: 1.64399 lat: 274.78864
IP:192.168.1.105 [1017 11:23:10 @model.py:254] Epoch[3] Batch[1200] Speed: 120.452097 samples/sec loss: 7.28725 acc: 0.38386 ce: 1.64292 lat: 273.36576
IP:192.168.1.105 [1017 11:23:37 @model.py:254] Epoch[3] Batch[1300] Speed: 119.701869 samples/sec loss: nan acc: 0.38344 ce: 1.64462 lat: nan
IP:192.168.1.105 [1017 11:24:03 @model.py:254] Epoch[3] Batch[1400] Speed: 119.566084 samples/sec loss: nan acc: 0.38032 ce: 1.65211 lat: nan
IP:192.168.1.105 [1017 11:24:31 @model.py:254] Epoch[3] Batch[1500] Speed: 117.215311 samples/sec loss: nan acc: 0.37724 ce: 1.65933 lat: nan
IP:192.168.1.105 [1017 11:24:46 @model.py:279] Start to train theta for epoch 4
IP:192.168.1.105 [1017 11:25:10 @model.py:254] Epoch[4] Batch[100] Speed: 81.351598 samples/sec loss: nan acc: 0.37242 ce: nan lat: nan
IP:192.168.1.105 [1017 11:25:34 @model.py:254] Epoch[4] Batch[200] Speed: 135.138710 samples/sec loss: nan acc: 0.36960 ce: nan lat: nan
IP:192.168.1.105 [1017 11:25:58 @model.py:254] Epoch[4] Batch[300] Speed: 129.706056 samples/sec loss: nan acc: 0.36681 ce: nan lat: nan
IP:192.168.1.105 [1017 11:26:23 @model.py:254] Epoch[4] Batch[400] Speed: 130.212272 samples/sec loss: nan acc: 0.36412 ce: nan lat: nan
IP:192.168.1.105 [1017 11:26:47 @model.py:254] Epoch[4] Batch[500] Speed: 130.657551 samples/sec loss: nan acc: 0.36137 ce: nan lat: nan
IP:192.168.1.105 [1017 11:27:12 @model.py:254] Epoch[4] Batch[600] Speed: 130.228069 samples/sec loss: nan acc: 0.35875 ce: nan lat: nan
IP:192.168.1.105 [1017 11:27:37 @model.py:254] Epoch[4] Batch[700] Speed: 128.510636 samples/sec loss: nan acc: 0.35617 ce: nan lat: nan
IP:192.168.1.105 [1017 11:28:01 @model.py:254] Epoch[4] Batch[800] Speed: 130.884915 samples/sec loss: nan acc: 0.35369 ce: nan lat: nan
IP:192.168.1.105 [1017 11:28:26 @model.py:254] Epoch[4] Batch[900] Speed: 131.872347 samples/sec loss: nan acc: 0.35116 ce: nan lat: nan
IP:192.168.1.105 [1017 11:28:50 @model.py:254] Epoch[4] Batch[1000] Speed: 131.942107 samples/sec loss: nan acc: 0.34866 ce: nan lat: nan
IP:192.168.1.105 [1017 11:29:14 @model.py:254] Epoch[4] Batch[1100] Speed: 132.766633 samples/sec loss: nan acc: 0.34630 ce: nan lat: nan
IP:192.168.1.105 [1017 11:29:38 @model.py:254] Epoch[4] Batch[1200] Speed: 132.858248 samples/sec loss: nan acc: 0.34405 ce: nan lat: nan
`
